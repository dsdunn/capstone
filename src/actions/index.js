export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const updateUser = (user) => ({
  type: 'UPDATE_USER',
  user
})

export const setProfile = profile => ({
  type: 'SET_PROFILE',
  profile
})

export const addCollections = (collections) => ({
  type: 'ADD_COLLECTIONS',
  collections
})

export const setCollection = collection => ({
  type: 'SET_COLLECTION',
  collection
})

export const updateCollectionsList = (collectionsList) => {
  type: 'UPDATE_COLLECTIONS_LIST',
  collectionsList
}