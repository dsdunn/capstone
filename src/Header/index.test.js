import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapDispatchToProps, mapStateToProps } from './'
import { signOut } from '../actions';

describe('Header', () => {
  let wrapper;
  let mockProfile;
  let mockSignOut = jest.fn()
  let mockHistory;

  beforeEach(() => {
    mockProfile = {
      avatar: '2 comes out in 2020',
      username: 'username? username for what?',
      location: 'somewhere else',
      bio: 'dome'
    }
    mockHistory = { 
                    location: { pathname: '/'},
                    goBack: jest.fn(),
                    push: jest.fn() 
                  }
    wrapper = shallow(<Header user={mockProfile}
                              signOut={mockSignOut}
                              history={mockHistory}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should start with default state of dashboardActive, and searchInput', () => {
    const expected = { 
      dashboardActive: false,
      searchInput: ''
    }

    const actual = wrapper.state()
    expect(actual).toEqual(expected)
  })

  it('should sign out when you call handleSignOut', () => {
    wrapper.instance().handleSignOut()
    expect(mockSignOut).toHaveBeenCalled()
  })

  it('should toggle dashboard state when handleDashboard is called', () => {
    wrapper.instance().handleDashboard()
    const actual = wrapper.state('dashboardActive')
    const expected = true

    expect(actual).toEqual(expected)
  })

  it('should call loginLocation when the Login btn is clicked', () => {
    const spy = jest.spyOn(wrapper.instance(), 'loginLocation')
    wrapper.find('.test1').simulate('click');

    expect(spy).toHaveBeenCalled();
  })

  it('should call signupLocation when the Login btn is clicked', () => {
    const spy = jest.spyOn(wrapper.instance(), 'signupLocation')
    wrapper.find('.test2').simulate('click');

    expect(spy).toHaveBeenCalled();
  })

  it('should update searchInput state when handleSearchChange is called', () => {
    const mockEvent = { target: { value: 'sick value dude!' } }
    wrapper.instance().handleSearchChange(mockEvent)
    const actual = wrapper.state('searchInput')
    const expected = 'sick value dude!' 

    expect(actual).toEqual(expected)
  })

  it('should return a props object with signIn', () => {
    const mockSignIn = jest.fn();
    const mockDispatch = jest.fn();

    const mappedProps = mapDispatchToProps(mockDispatch)
    const actionToDispatch = signOut();

    mappedProps.signOut();
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('should return a props object', () => {
    const mockState = {user: {}};
    const expected = {"user": {}};

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })
})




























