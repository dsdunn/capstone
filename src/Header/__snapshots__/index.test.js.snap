// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    history={
        Object {
            "goBack": [Function],
            "location": Object {
              "pathname": "/",
            },
            "push": [Function],
          }
    }
    signOut={[Function]}
    user={
        Object {
            "avatar": "2 comes out in 2020",
            "bio": "dome",
            "location": "somewhere else",
            "username": "username? username for what?",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
>
          <Link
                    replace={false}
                    to="/home"
          >
                    <h1
                              className="header-app-name"
                    >
                              Collec
                              <span>
                                        share
                              </span>
                    </h1>
          </Link>
          <input
                    className="header-search-bar"
                    onChange={[Function]}
                    placeholder="Search"
                    type="text"
                    value=""
          />
          <div
                    className="header-nav"
          >
                    <div>
                              <a
                                        className="header-link test1"
                                        onClick={[Function]}
                              >
                                        Login
                              </a>
                              <a
                                        className="header-link test2"
                                        onClick={[Function]}
                              >
                                        Create Account
                              </a>
                              <Link
                                        className="header-link"
                                        replace={false}
                                        to="/home"
                              >
                                        Get All
                              </Link>
                    </div>
          </div>
</div>,
        <withRouter(Connect(Component))
          active={false}
          handleSignOut={[Function]}
/>,
      ],
      "className": "header-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/home"
>
              <h1
                            className="header-app-name"
              >
                            Collec
                            <span>
                                          share
                            </span>
              </h1>
</Link>,
            <input
              className="header-search-bar"
              onChange={[Function]}
              placeholder="Search"
              type="text"
              value=""
/>,
            <div
              className="header-nav"
>
              <div>
                            <a
                                          className="header-link test1"
                                          onClick={[Function]}
                            >
                                          Login
                            </a>
                            <a
                                          className="header-link test2"
                                          onClick={[Function]}
                            >
                                          Create Account
                            </a>
                            <Link
                                          className="header-link"
                                          replace={false}
                                          to="/home"
                            >
                                          Get All
                            </Link>
              </div>
</div>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h1
                className="header-app-name"
>
                Collec
                <span>
                                share
                </span>
</h1>,
              "replace": false,
              "to": "/home",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Collec",
                  <span>
                    share
</span>,
                ],
                "className": "header-app-name",
              },
              "ref": null,
              "rendered": Array [
                "Collec",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "share",
                  },
                  "ref": null,
                  "rendered": "share",
                  "type": "span",
                },
              ],
              "type": "h1",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "header-search-bar",
              "onChange": [Function],
              "placeholder": "Search",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                <a
                                className="header-link test1"
                                onClick={[Function]}
                >
                                Login
                </a>
                <a
                                className="header-link test2"
                                onClick={[Function]}
                >
                                Create Account
                </a>
                <Link
                                className="header-link"
                                replace={false}
                                to="/home"
                >
                                Get All
                </Link>
</div>,
              "className": "header-nav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    className="header-link test1"
                    onClick={[Function]}
>
                    Login
</a>,
                  <a
                    className="header-link test2"
                    onClick={[Function]}
>
                    Create Account
</a>,
                  <Link
                    className="header-link"
                    replace={false}
                    to="/home"
>
                    Get All
</Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                    "className": "header-link test1",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Create Account",
                    "className": "header-link test2",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Create Account",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Get All",
                    "className": "header-link",
                    "replace": false,
                    "to": "/home",
                  },
                  "ref": null,
                  "rendered": "Get All",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": false,
          "handleSignOut": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
>
            <Link
                        replace={false}
                        to="/home"
            >
                        <h1
                                    className="header-app-name"
                        >
                                    Collec
                                    <span>
                                                share
                                    </span>
                        </h1>
            </Link>
            <input
                        className="header-search-bar"
                        onChange={[Function]}
                        placeholder="Search"
                        type="text"
                        value=""
            />
            <div
                        className="header-nav"
            >
                        <div>
                                    <a
                                                className="header-link test1"
                                                onClick={[Function]}
                                    >
                                                Login
                                    </a>
                                    <a
                                                className="header-link test2"
                                                onClick={[Function]}
                                    >
                                                Create Account
                                    </a>
                                    <Link
                                                className="header-link"
                                                replace={false}
                                                to="/home"
                                    >
                                                Get All
                                    </Link>
                        </div>
            </div>
</div>,
          <withRouter(Connect(Component))
            active={false}
            handleSignOut={[Function]}
/>,
        ],
        "className": "header-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/home"
>
                <h1
                                className="header-app-name"
                >
                                Collec
                                <span>
                                                share
                                </span>
                </h1>
</Link>,
              <input
                className="header-search-bar"
                onChange={[Function]}
                placeholder="Search"
                type="text"
                value=""
/>,
              <div
                className="header-nav"
>
                <div>
                                <a
                                                className="header-link test1"
                                                onClick={[Function]}
                                >
                                                Login
                                </a>
                                <a
                                                className="header-link test2"
                                                onClick={[Function]}
                                >
                                                Create Account
                                </a>
                                <Link
                                                className="header-link"
                                                replace={false}
                                                to="/home"
                                >
                                                Get All
                                </Link>
                </div>
</div>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h1
                  className="header-app-name"
>
                  Collec
                  <span>
                                    share
                  </span>
</h1>,
                "replace": false,
                "to": "/home",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Collec",
                    <span>
                      share
</span>,
                  ],
                  "className": "header-app-name",
                },
                "ref": null,
                "rendered": Array [
                  "Collec",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "share",
                    },
                    "ref": null,
                    "rendered": "share",
                    "type": "span",
                  },
                ],
                "type": "h1",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "header-search-bar",
                "onChange": [Function],
                "placeholder": "Search",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  <a
                                    className="header-link test1"
                                    onClick={[Function]}
                  >
                                    Login
                  </a>
                  <a
                                    className="header-link test2"
                                    onClick={[Function]}
                  >
                                    Create Account
                  </a>
                  <Link
                                    className="header-link"
                                    replace={false}
                                    to="/home"
                  >
                                    Get All
                  </Link>
</div>,
                "className": "header-nav",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      className="header-link test1"
                      onClick={[Function]}
>
                      Login
</a>,
                    <a
                      className="header-link test2"
                      onClick={[Function]}
>
                      Create Account
</a>,
                    <Link
                      className="header-link"
                      replace={false}
                      to="/home"
>
                      Get All
</Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Login",
                      "className": "header-link test1",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": "a",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Create Account",
                      "className": "header-link test2",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Create Account",
                    "type": "a",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Get All",
                      "className": "header-link",
                      "replace": false,
                      "to": "/home",
                    },
                    "ref": null,
                    "rendered": "Get All",
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "handleSignOut": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
